### PR TITLE
Schedule Based Airspace

### PR DESCRIPTION
**Pull request by Michael-AirMap**

### CODE REVIEW

File: Source/Core/Models/Advisory/AirMapAdvisory.swift [hunk 1]
```diff
@@ -202,6 +205,79 @@ public struct AirMapAdvisory {
 		public let effective: Date?
 		public let size: Hectares?
 	}
+
+
+	/// A time sheet to describe when the airspace is active
+	public struct Timesheet {
+		public let active: Bool?
```
[the remark this answers is not in the document]
Comment (Michael-AirMap) [reply Source/Core/Models/Advisory/AirMapAdvisory.swift:213]: Done

File: Source/Core/Models/Advisory/AirMapAdvisory.swift [hunk 2]
```diff
@@ -202,6 +205,79 @@ public struct AirMapAdvisory {
...
+			case busyFriday = "day_busy_friday"
+			case any = "day_any"
+		}
+
+		public enum Event: String, CaseIterable {
+			case sunrise = "event_sunrise"
+			case sunset = "event_sunset"
+		}
+
+		public enum EventInterpretationDescriptor: String {
```
[the remark this answers is not in the document]
Comment (Michael-AirMap) [reply Source/Core/Models/Advisory/AirMapAdvisory.swift:284]: Done

File: Source/Core/Models/EquatableHashableComparable.swift [hunk 3]
```diff
@@ -166,3 +166,99 @@ extension AirMapFlightFeature.Status: Comparable {
...
 }
+
+extension AirMapAdvisory.Timesheet.DayDescriptor: Equatable, Hashable, Comparable {
+
+	public static func ==(lhs: AirMapAdvisory.Timesheet.DayDescriptor, rhs: AirMapAdvisory.Timesheet.DayDescriptor) -> Bool {
+		return lhs.hashValue == rhs.hashValue
+	}
+
+	public func hash(into hasher: inout Hasher) {
+		hasher.combine(name.hashValue)
```
[the remark this answers is not in the document]
Comment (Michael-AirMap) [reply Source/Core/Models/EquatableHashableComparable.swift]: Updated

File: Source/UI/Extensions/CharacterSet+AirMap.swift [hunk 4]
```diff
@@ -0,0 +1,32 @@
+//
+//  CharacterSet+AirMap.swift
+//  AirMapSDK
+//
+//  Created by Michael Odere on 4/9/20.
+//  Authored by Adolfo Martinelli
```
[the remark this answers is not in the document]
Comment (Michael-AirMap) [reply Source/UI/Extensions/CharacterSet+AirMap.swift:6]: I don't want to steal credit 😄. To be clear omit the suggested change and the "Authored by" line or just the suggested change?

File: Source/Core/Models/EquatableHashableComparable.swift [hunk 5]
```diff
@@ -166,3 +166,99 @@ extension AirMapFlightFeature.Status: Comparable {
 		return lhs.order < rhs.order
 	}
 }
+
+extension AirMapAdvisory.Timesheet.DayDescriptor: Equatable, Hashable, Comparable {
+
+	public static func ==(lhs: AirMapAdvisory.Timesheet.DayDescriptor, rhs: AirMapAdvisory.Timesheet.DayDescriptor) -> Bool {
+		return lhs.hashValue == rhs.hashValue
+	}
```
[the remark this answers is not in the document]
Comment (Michael-AirMap) [reply Source/Core/Models/EquatableHashableComparable.swift:174]: It looks like it does not autogenerate in extensions
`Implementation of 'Equatable' cannot be automatically synthesized in an extension in a different file to the type`

File: Source/Core/Models/Advisory/AirMapAdvisory.swift [hunk 6]
```diff
@@ -56,6 +56,9 @@ public struct AirMapAdvisory {
 	/// Additional metadata specific to the advisory type
 	public let properties: AdvisoryProperties?
 
+	/// Describes the list of intervals the advisory is active
+	public let timeSheets: [Timesheet]?
```
[the remark this answers is not in the document]
Comment (Michael-AirMap) [reply Source/Core/Models/Advisory/AirMapAdvisory.swift]: Fixed

File: Source/UI/Views/AirMapMapView.swift [hunk 7]
```diff
@@ -79,6 +79,24 @@ open class AirMapMapView: MGLMapView {
 	public enum TemporalRange {
 		case sliding(window: TimeInterval)
 		case fixed(start: Date, end: Date)
+
+		var startDate: Date {
```
[the remark this answers is not in the document]
Comment (Michael-AirMap) [reply Source/UI/Views/AirMapMapView.swift]: Sure. Should we rename `endDate` to `effectiveEnd`?

File: Source/Core/Models/Advisory/AirMapAdvisory.swift [hunk 8]
```diff
@@ -202,6 +205,79 @@ public struct AirMapAdvisory {
...
 	}
+
+
+	/// A time sheet to describe when the airspace is active
+	public struct Timesheet {
+		public let active: Bool?
+		public let timesheetData: Data?
+
+		public struct Data {
+			public let offsetUTC: Int?
```
[the remark this answers is not in the document]
Comment (Michael-AirMap) [reply Source/Core/Models/Advisory/AirMapAdvisory.swift:219]: Done

File: Source/UI/Extensions/CharacterSet+AirMap.swift (hunk 4 above)
[the remark this answers is not in the document]
Comment (adolfo) [reply Source/UI/Extensions/CharacterSet+AirMap.swift:6]: Just omit the `Authored by` line. No need to commit the suggestion.

File: Source/UI/Views/AirMapMapView.swift (hunk 7 above)
[the remark this answers is not in the document]
Comment (adolfo) [reply Source/UI/Views/AirMapMapView.swift]: Yes

**Pull request merged**